src/metabase/pulse.clj

```diff
@@ -232,28 +232,31 @@
     true))
 
 (defn- get-notification-info
-  [pulse parts channel]
+  [pulse parts pulse-channel]
   (let [alert? (nil? (:dashboard_id pulse))]
-    (merge {:payload-type (if alert?
-                            :notification/alert
-                            :notification/dashboard-subscription)
-            :payload      (if alert? (first parts) parts)
-            :pulse        pulse
-            :channel      channel}
+    (merge {:payload-type  (if alert?
+                             :notification/alert
+                             :notification/dashboard-subscription)
+            :payload       (if alert? (first parts) parts)
+            :pulse         pulse
+            :pulse-channel pulse-channel}
            (if alert?
              {:card  (t2/select-one :model/Card (-> parts first :card :id))}
              {:dashboard (t2/select-one :model/Dashboard (:dashboard_id pulse))}))))
 
-(defn- channels-to-channel-recipients
-  [channel]
-  (if (= :slack (keyword (:channel_type channel)))
-    [(get-in channel [:details :channel])]
-    (for [recipient (:recipients channel)]
+(defn- channel-recipients
+  [pulse-channel]
+  (case (keyword (:channel_type pulse-channel))
+    :slack
+    [(get-in pulse-channel [:details :channel])]
+    :email
+    (for [recipient (:recipients pulse-channel)]
       (if-not (:id recipient)
         {:kind :external-email
          :email (:email recipient)}
         {:kind :user
-         :user recipient}))))
+         :user recipient}))
+    []))
```

Comment:
I think it's worth logging a warning in the unknown case.

Reply:
log a warning here 3ba4207d92